applications/asset_tracker_v2/src/events/app_module_event.c

```diff
@@ -61,23 +61,32 @@ static char *get_evt_type_str(enum app_module_event_type type)
 	}
 }
 
+#define LOG_EVENT_BUF_SIZE 50
 static void log_event(const struct app_event_header *aeh)
 {
 	const struct app_module_event *event = cast_app_module_event(aeh);
-	char data_types[50] = "\0";
+	uint16_t size = 0;
+	char data_types[LOG_EVENT_BUF_SIZE] = "\0";
 
 	if (event->type == APP_EVT_ERROR) {
 		APP_EVENT_MANAGER_LOG(aeh, "%s - Error code %d",
 				get_evt_type_str(event->type), event->data.err);
 	} else if (event->type == APP_EVT_DATA_GET) {
 		for (int i = 0; i < event->count; i++) {
-			strcat(data_types, type2str(event->data_list[i]));
+			strncat(data_types, type2str(event->data_list[i]),
+					LOG_EVENT_BUF_SIZE - size - 1);
+			size += strlen(type2str(event->data_list[i]));
 
-			if (i == event->count - 1) {
+			if (i == event->count - 1 || size > LOG_EVENT_BUF_SIZE - 2) {
```

Comment:
Although the change looks sane, there are some magic numbers that can be difficult to understand at first glance. Typically I feel that any subtraction of 1 is understandable when dealing with string parsing. But, subtracting `2` can be a little more difficult to understand the meaning of. I suggest creating a define with a comment that explains its purpose, and using the macro in code instead.

Reply:
I changed the comparison structure so that it's hopefully clearer, and also added a comment to explain why the comparison is as is.